packages/payload/src/fields/setDefaultBeforeDuplicate.ts

```diff
@@ -18,16 +19,18 @@ export const setDefaultBeforeDuplicate = (
     (!field.hooks?.beforeDuplicate ||
       (Array.isArray(field.hooks.beforeDuplicate) && field.hooks.beforeDuplicate.length === 0))
   ) {
-    if ((field.type === 'text' || field.type === 'textarea') && field.required && field.unique) {
-      field.hooks.beforeDuplicate = [
-        fieldShouldBeLocalized({ field, parentIsLocalized })
-          ? localizedUniqueRequired
-          : uniqueRequired,
-      ]
-    } else if (field.unique) {
-      field.hooks.beforeDuplicate = [
-        fieldShouldBeLocalized({ field, parentIsLocalized }) ? localizedUnique : unique,
-      ]
+    if (field.unique) {
+      const updateStrategy = handleUniqueFields(field.type)
+
+      const strategyHooks = {
+        append: [fieldShouldBeLocalized({ field, parentIsLocalized }) ? localizedUnique : unique],
+        numericallyAppend: [numberUnique],
+        undefined: [resetUnique],
+      }
+
+      if (strategyHooks[updateStrategy]) {
+        field.hooks.beforeDuplicate = strategyHooks[updateStrategy]
+      }
```

Comment:
Here I think we should just inline the checks instead of using the helper function, just so it is easier to read. Can just be something like:

```ts
if (['email', 'point', 'relationship', 'select', 'upload', 'number'].includes(field.type)) {
  field.hooks.beforeDuplicate = [() => undefined]
} else if (['code', 'json', 'text', 'textarea'].includes(field.type)) {
  field.hooks.beforeDuplicate = fieldShouldBeLocalized({ field, parentIsLocalized }) ? [localizedUnique] : [unique]
}
```

Reply:
Updated